x-pack/plugins/security_solution/public/detections/components/alerts_kpis/common/hooks.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import type { GlobalTimeArgs } from '../../../../common/containers/use_global_time';
+
+interface UseInspectButtonParams extends Pick<GlobalTimeArgs, 'setQuery' | 'deleteQuery'> {
+  response: string;
+  request: string;
+  refetch: (() => void) | null;
+  uniqueQueryId: string;
+  loading: boolean;
+}
+/**
+ * * Add query to inspect button utility.
+ * * Delete query from inspect button utility when component unmounts
+ */
+export const useInspectButton = ({
+  setQuery,
+  response,
+  request,
+  refetch,
+  uniqueQueryId,
+  deleteQuery,
+  loading,
+}: UseInspectButtonParams) => {
+  useEffect(() => {
+    if (refetch != null && setQuery != null) {
+      setQuery({
+        id: uniqueQueryId,
+        inspect: {
+          dsl: [request],
+          response: [response],
+        },
+        loading,
+        refetch,
+      });
+    }
+  }, [setQuery, loading, response, request, refetch, uniqueQueryId]);
+
+  useEffect(() => {
+    return () => {
+      if (deleteQuery) {
+        deleteQuery({ id: uniqueQueryId });
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is this needed?

Reply:
I moved this code from some other place. From what I understand, it will delete the query that `setQuery` adds when rendering the component. So you have a good point: "It doesn't do anything on re-renders". Maybe it would be clearer if I move both of them to the same `useEffect`. It would better illustrate the connection between both calls.

Let me implement it!  👍🏽